src/main/java/Parser.java

```diff
@@ -0,0 +1,150 @@
+                throw new DukeException("OOPS!!! The description of a todo cannot be empty.");
+            }
+            String todoTask = getTodoTask(input);
+
+            TaskList.createTodoTask(todoTask);
+
+            return ui.showMessage("added!\n" + "you have " + TaskList.updatedTaskList.size()
+                    + " tasks in your list");
+        } else if (input.startsWith("event ")) {
+            String eventTask = getEventTask(input);
+            String date = getEventDate(input);
+
+            TaskList.createEventTask(eventTask, date);
+
+            return ui.showMessage("added!\n" + "you have " + TaskList.updatedTaskList.size()
+                    + " tasks in your list");
+        } else if (input.startsWith("deadline ")) {
+            String deadlineTask = getDeadlineTask(input);
+            LocalDate deadline = getDeadlineDate(input);
+
+            TaskList.createDeadlineTask(deadlineTask, deadline);
+
+            return ui.showMessage("added!\n" + "you have " + TaskList.updatedTaskList.size()
+                    + " tasks in your list");
+        } else if (input.startsWith("delete ")) {
+            int taskNumber = getTaskNumberToBeDeleted(input);
+            Task taskDeleted = getTaskToBeDeleted(taskNumber);
+
+            TaskList.deleteTask(taskNumber);
+
+            return ui.showMessage("Noted. I've removed this task: " + taskDeleted + "\n"
+                    + "you have " + TaskList.updatedTaskList.size() + " tasks in your list");
+        } else {
+            throw new DukeException("OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    public static String getKeyword(String input) {
+        return input.substring(5);
+    }
+
+    /**
+     * Returns the completed task from the task list
+     * @param input supplied by the user
+     * @return the task description of the completed task
+     */
+    public static Task getFinishedTask(String input) {
+        int taskNumber = Integer.parseInt(input.substring(5));
+        return TaskList.updatedTaskList.get(taskNumber - 1);
+    }
+
+    /**
+     * Returns the task number to be deleted upon filtering the input
+     * @param input supplied by the user
+     * @return task number to be deleted
+     */
+    public static int getTaskNumberToBeDeleted(String input) {
+        return Integer.parseInt(input.substring(7));
+    }
+
+    /**
+     * Returns the task to be deleted from the task list
+     * @param taskNumber of the task (in the task list) to be deleted
+     * @return the task to be deleted
+     */
+    public static Task getTaskToBeDeleted(int taskNumber) {
+        return TaskList.updatedTaskList.get(taskNumber - 1);
+    }
+
+    /**
+     * Returns the todo task description upon filtering the input
+     * @param input supplied by the user
+     * @return todo task description
+     */
+    public static String getTodoTask(String input) {
+        return input.substring(5);
```

Comment:
Perhaps the use of constants such as `TODO_TASK_INDEX` instead of "magic numbers" would be better?